src/app/admin/websiteDataTable/WebsiteDataTable.tsx

```diff
@@ -0,0 +1,231 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import supabase from "@/supabaseClient";
+import {
+    DataGrid,
+    GridActionsCellItem,
+    GridColDef,
+    GridEventListener,
+    GridRowEditStopReasons,
+    GridRowId,
+    GridRowModes,
+    GridRowModesModel,
+    useGridApiRef,
+} from "@mui/x-data-grid";
+import EditIcon from "@mui/icons-material/Edit";
+import SaveIcon from "@mui/icons-material/Save";
+import CancelIcon from "@mui/icons-material/Close";
+import { LinearProgress } from "@mui/material";
+import { DatabaseError } from "@/app/errorClasses";
+import { fetchWebsiteData, updateDbWebsiteData } from "./FetchWebsiteData";
+import CustomComponent from "./CustomEditComponent";
+import { logErrorReturnLogId, logInfoReturnLogId } from "@/logger/logger";
+
+export interface WebsiteDataRow {
+    readableName: string;
+    id: string;
+    value: string;
+}
+
+const WebsiteDataTable: React.FC = () => {
+    const [rows, setRows] = useState<WebsiteDataRow[]>([]);
+    const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({});
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const apiRef = useGridApiRef();
+
+    useEffect(() => {
+        fetchWebsiteData()
+            .then((response) => setRows(response))
+            .catch(async (error) => {
+                const logId = await logErrorReturnLogId("Error with fetch: website data", error);
+                throw new DatabaseError("fetch", "website data table", logId);
+            })
+            .finally(() => setIsLoading(false));
+    }, []);
+
+    useEffect(() => {
+        // This requires that the DB table has Realtime turned on
+        const subscriptionChannel = supabase
+            .channel("website-data-table-changes")
+            .on(
+                "postgres_changes",
+                { event: "*", schema: "public", table: "website_data" },
+                async () => {
+                    try {
+                        const websiteData = await fetchWebsiteData();
+                        setRows(websiteData);
+                    } catch (error) {
+                        if (error) {
+                            const logId = await logErrorReturnLogId(
+                                "Error with fetch: website data subscription",
+                                error
+                            );
+                            setRows([]);
+                            throw new DatabaseError("fetch", "website data table", logId);
```

Comment:
I don't think we'll be able to catch this error - I think replacing this with displaying an error message will be fine